packages/components/src/CalendarDatePicker/components/CalendarDatePickerGrid.tsx

```diff
@@ -0,0 +1,217 @@
+  startOfMonth,
+  startOfWeek,
+} from "../utils";
+import { useGridKeyboardControl } from "../hooks/useGridKeyboardControl";
+import { useSyncFocusAndViewingDate } from "../hooks/useSyncFocusAndViewingDate";
+import { useOnToggleDate } from "../hooks/useOnToggleDate";
+import { useHighlightedDatesGroupedByTimeStamp } from "../hooks/useHighlightedDatesGroupedByTimeStamp";
+
+interface CalendarDatePickerGridProps {
+  readonly selected?: Date[];
+  readonly viewingDate: Date;
+  readonly minDate?: Date;
+  readonly maxDate?: Date;
+  readonly highlightedDates: Date[];
+  readonly weekStartsOnMonday: boolean;
+  readonly translations?: {
+    readonly highlightedLabelSuffix?: string;
+    readonly chooseDate?: string;
+  };
+  readonly range: boolean;
+  readonly onChange?: (
+    date: Date[],
+    method: "click" | "enter" | "space",
+  ) => void;
+  readonly onMonthChange?: (date: Date) => void;
+}
+
+export const CalendarDatePickerGrid = ({
+  selected = [],
+  viewingDate,
+  highlightedDates = [],
+  minDate,
+  maxDate,
+  range,
+  weekStartsOnMonday,
+  translations,
+  onChange,
+  onMonthChange,
+}: CalendarDatePickerGridProps): JSX.Element => {
+  const [focusedDate, setFocusedDate] = useState<Date>(
+    selected[0] || viewingDate,
+  );
+
+  const onToggle = useOnToggleDate({
+    selected,
+    range,
+    onChange,
+    setFocusedDate,
+  });
+
+  const grid = useRowsAndCells({
+    selected,
+    minDate,
+    maxDate,
+    highlightedDates,
+    weekStartsOnMonday,
+    range,
+    viewingDate,
+    focusedDate,
+    highlightedLabelSuffix: translations?.highlightedLabelSuffix,
+    onToggle,
+  });
+
+  const onKeyDown = useGridKeyboardControl({
+    setFocusedDate,
+    onToggle,
+  });
+
+  useSyncFocusAndViewingDate({
+    viewingDate,
+    focusedDate,
+    setFocusedDate,
+    onMonthChange,
+  });
+
+  const id = useRef(`calendar-${Math.random()}`).current;
+
+  return (
+    <div
+      role="grid"
+      onKeyDown={onKeyDown}
+      id={id}
+      aria-label={translations?.chooseDate || "Choose date"}
+    >
+      {grid}
+    </div>
+  );
+};
+
+// eslint-disable-next-line max-statements
+function useRowsAndCells({
+  viewingDate,
+  selected,
+  minDate,
+  maxDate,
+  highlightedDates,
+  weekStartsOnMonday,
+  highlightedLabelSuffix,
+  range,
+  focusedDate,
+  onToggle,
+}: {
+  viewingDate: Date;
+  selected: Date[];
+  minDate?: Date;
+  maxDate?: Date;
+  highlightedDates: Date[];
+  weekStartsOnMonday: boolean;
+  highlightedLabelSuffix: string | undefined;
+  range: boolean;
+  focusedDate: Date;
+  onToggle: (
+    date: Date,
+    method: "click" | "enter" | "space",
+    isSelected?: boolean,
+  ) => void;
+}) {
+  const rows = [
+    <DaysOfTheWeekRow key="weekdays" weekStartsOnMonday={weekStartsOnMonday} />,
+  ];
+
+  const mapOfHighlightedDates =
+    useHighlightedDatesGroupedByTimeStamp(highlightedDates);
+
+  const { currentDate, month, lastDateOfTheMonth, startDate } = useMemo(() => {
```

Comment:
I'm not sure if this is of assistance, but in some recent calendar builds, I used `date-fns` to make the entire month view with a couple of helpers, which may allow for a simplification of building the calendar display. There's a great video from Sam here that also shows it off. https://youtu.be/9ySmMd5Cjc0?list=PLlNdnoKwDZdxdh22NW9r3YAoMsM-xaSeb&t=284

Reply:
Nice! I've avoided using third-party dependencies though that might conflict with any projects this is used in since this is an Atlantis component

Reply:
I only suggested it since `date-fns` is already in the project and used by your components. I thought it might simplify needing to do a lot of the date math for rows/start of week/etc.

Reply:
It's in the project but `@jobber/components` itself does not have it defined as a dependency

Reply:
Apologies. You're right.